trading_calendars/exchange_calendar_xist.py

```diff
@@ -0,0 +1,208 @@
+
+EidAlAdha4 = eid_al_adha_first_day + timedelta(3)
+
+VictoryDay = Holiday(
+    'Victory Day',
+    month=8,
+    day=30,
+)
+
+RepublicDay = Holiday(
+    'Republic Day',
+    month=10,
+    day=29,
+)
+
+# Early Closes
+DayBeforeEidAlFitr = eid_al_fitr_first_day - timedelta(1)
+
+EidAlFitrHalfDay = [
+    day for day in DayBeforeEidAlFitr if day.weekday() not in WEEKENDS
+]
+
+DayBeforeEidAlAdha = eid_al_adha_first_day - timedelta(1)
+
+EidAlAdhaHalfDay = [
+    day for day in DayBeforeEidAlAdha if day.weekday() not in WEEKENDS
+]
+
+RepublicDayHalfDay = Holiday(
+    'Republic Day half day',
+    month=10,
+    day=28,
+    days_of_week=WEEKDAYS,
+)
+
+
+class XISTExchangeCalendar(TradingCalendar):
+    """
+    Exchange calendar for the Istanbul Stock Exchange (XIST).
+
+    # Note: Effective as of Nov 14, 2016 according to official website
+    Open Time: 10:00 AM, TRT
+    Close Time: 6:00 PM, TRT
+
+    Regularly-Observed Holidays:
+    - New Year's Day
+    - National Sovereignty and Children's Day (Apr 23)
+    - Labour Day (May 1)
+    - Commemoration of Attaturk, Youth and Sports Day (May 19)
+    - Eid-ul-Fitr (Islamic Lunar Calendar)
+    - Democracy and National Unity Day (Jul 15)
+    - Eid-al-Adha (first day is a half-day, remaining full holidays)
+    - Victory Day (Aug 30)
+    - Republic Day (Oct 29)
+
+    Early Closes:
+    - First Day of Eid-al-Adha
+    - Republic Day (Oct 28)
+    """
+    name = 'XIST'
+
+    tz = timezone('Europe/Istanbul')
+
+    open_times = (
+        (None, time(10, 1)),
+    )
+
+    close_times = (
+        (None, time(18, 00)),
+    )
+
+    regular_early_close = time(12, 30)
+
+    @property
+    def regular_holidays(self):
+        return HolidayCalendar([
+            NewYearsDay,
+            NationalSovereigntyAndChildrensDay,
+            LabourDay,
+            CommemorationOfAttaturkYouthAndSportsDay,
+            DemocracyAndNationalUnityDay,
+            VictoryDay,
+            RepublicDay,
+        ])
+
+    @property
+    def adhoc_holidays(self):
+        misc_adhoc = [
+            pd.Timestamp('2002-01-04', tz=UTC),  # Market Holiday
+            pd.Timestamp('2003-02-14', tz=UTC),  # Eid al Adha extra holiday
+            pd.Timestamp('2003-11-21', tz=UTC),  # Terror attacks
+            pd.Timestamp('2003-11-24', tz=UTC),  # Eid al Fitr extra holiday
+            pd.Timestamp('2003-11-28', tz=UTC),  # Eid al Fitr extra holiday
+            pd.Timestamp('2004-01-23', tz=UTC),  # Bad weather
+            pd.Timestamp('2004-12-30', tz=UTC),  # Closure for redenomination
+            pd.Timestamp('2004-12-31', tz=UTC),  # Closure for redenomination
+            pd.Timestamp('2006-01-13', tz=UTC),  # Eid al Adha extra holiday
+        ]
+
+        return list(chain(
+            misc_adhoc,
+            EidAlFitr1,
+            EidAlFitr2,
+            EidAlFitr3,
+            EidAlAdha1,
+            EidAlAdha2,
+            EidAlAdha3,
+            EidAlAdha4,
```

Comment:
Do we care that these could include weekends? I don't think so, but just double checking.

Reply:
I don't think we do in this case, because these are defining full holidays so if they fall on a weekend the redundancy shouldn't matter.  We do care when defining early closures, because it doesn't make sense to say that there was an early closure on a holiday.